### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by Scron-Chang**
Valgrind shows:
Enabled options:
   --leak-check=full
   --show-leak-kinds=all
   --track-origins=yes
   --verbose
HEAP SUMMARY:
    in use at exit: 0 bytes in 0 blocks
  total heap usage: 15,772 allocs, 15,772 frees, 178,310,738 bytes allocated

The problem is found in functions: do_bic_update and send_recv_command
when invoking the freeipmi APIs.

Tested:
Checked by valgrind, and the result shows:
All heap blocks were freed -- no leaks are possible

Signed-off-by: Scron Chang <Scron.Chang@quantatw.com>

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -524,6 +531,9 @@ static int do_bic_update(uint8_t *buff, uint32_t buff_len)
             break;
     }
 
+clean:
+    ipmi_ctx_close (ipmi_ctx);
+    ipmi_ctx_destroy (ipmi_ctx);
     return 0;
```
Comment (MouchenHung) [inline src/main.c]: should be "return ret"

File: src/main.c [hunk 2]
```diff
@@ -524,6 +531,9 @@ static int do_bic_update(uint8_t *buff, uint32_t buff_len)
             break;
     }
```
Comment (MouchenHung) [inline src/main.c:532]: add "ret = 0" here

File: src/main.c [hunk 3]
```diff
@@ -513,8 +518,10 @@ static int do_bic_update(uint8_t *buff, uint32_t buff_len)
                 log_print(LOG_WRN, "Given update offset not mach with previous record!\n");
                 log_print(LOG_NON, "         Retry in few seconds...\n");
             }
-            return 1;
+            ret = 1;
```
Comment (MouchenHung) [inline src/main.c]: remove "ret = 1"

File: src/main.c [hunk 4]
```diff
@@ -513,8 +518,10 @@ static int do_bic_update(uint8_t *buff, uint32_t buff_len)
...
+            goto clean;
         }
+        ret = 0;
```
Comment (MouchenHung) [inline src/main.c]: remove "ret = 0"

**Review by MouchenHung — CHANGES_REQUESTED** [review]
some modify

File: src/main.c (hunk 4 above)
Comment (Scron-Chang) [reply src/main.c]: Done

File: src/main.c (hunk 3 above)
Comment (Scron-Chang) [reply src/main.c]: Done

File: src/main.c (hunk 2 above)
Comment (Scron-Chang) [reply src/main.c:532]: Done

File: src/main.c (hunk 1 above)
Comment (Scron-Chang) [reply src/main.c]: Done

**Pull request merged**